web/static/js/components/thesis-editor.js

```diff
@@ -176,25 +158,16 @@ class ThesisEditor extends React.Component {
     }
   }
 
-  postToServer (page, contents) {
-    Net.put('/thesis/update', {page, contents}).then((resp) => {
-      if (page.slug !== window.location.pathname) {
-        window.location = page.slug
-      } else {
-        this.setState({editing: false, pageModified: false, trayOpen: false})
-        this.setState({pageToolsHidden: true})
-      }
-    }).catch((err) => {
-      window.alert(err)
+  save (page, contents) {
+    this.external.save(page, contents, () => {
```

Comment:
So instead of containing the `Net` logic in here, we call `this.external.save()` to save it, and provide a callback function on success.